### DISCUSSION TITLE
'new' slug in dynamic route parameter.

### DISCUSSION THREAD

**Discussion by owen2jan** [nuxt/nuxt#26233] [Questions] [answered]
I have a route posts/[slug].vue. If I have a post with the slug 'new' the route object fails. i.e., 

```
const route = useRoute()
console.log(route)
//logs nothing
```

are there keywords that should not be used

**Comment by owen2jan** [comment]
Never mind. I have another page posts/new. I get redirected there instead. The two pages have a similar empty layout for testing that's why I was getting confused. What can I do in a case where the slug has a value similar to another page in same directory. i.e.
~~~
/posts
->/[slug].vue
->new.vue
~~~

**Reply by TurtleKwitty** [reply, marked as answer]
Static routes will always be picked ahead of wildcard routes; you'll need to move the [slug].vue to another route where ambiguity is impossible if that's what you required, something like /posts/bySlug/[slug].vue or the opposite by moving the new.vue page somewhere that doesnt conflict with the wildcard /admin/post/new.vue for example

**Comment by xjccc** [comment]
You can use definePageMeta path to custom regular.

[custom-regular-expression](https://nuxt.com/docs/api/utils/define-page-meta#using-a-custom-regular-expression)

```
<script setup lang="ts">
definePageMeta({
  path: '/:slug(^(?!new$).+)' 
})
</script>
```

And there has a [nuxt tip](https://www.youtube.com/embed/SXk-L19gTZk)

**Reply by owen2jan** [reply]
Can you explain what the regex '/:slug(^(?!new$).+)' does? I'm a bit confused.

**Reply by xjccc** [reply]
I think what you want is, /new -> new.vue, /others -> [slug].vue. So, the regex is not `new` to match [slug]. This is?

**Reply by owen2jan** [reply]
Thanks, I was trying to avoid @TurtleKwitty  's workaround but it seems I need another directory to put the new page so they don't conflict. I had some SEO concerns with this method however the new page will only be accessible in the admin section which doesn't need SEO.
